### ISSUE TITLE
Potential issues from location.href data passed to xhr.send

### ISSUE THREAD

**Issue by nettrino** [crashappsec/www-qa-issues#23]
The following snippet reads from `location.href` and passes unsanitized data to `xhr.send`. This could potentially open up DOM-based vulnerabilities like DOM-based open redirects.

```javascript
var perfData = {
   url: location.href,
   portal: 25939634,
   content: 52538649036,
   group: -1,
   connection: networkInfo,
   timing: performance.timing
   };
   xhr.send(JSON.stringify(perfData));
```

**Comment by curphey** [comment]
Makes sense. @nettrino wont the HTTP headers we are going to set https://github.com/crashappsec/www-qa-issues/issues/11 mitigate this ?

**Comment by curphey** [comment]
Theo, can you take a look at this for us please.

**Comment by nettrino** [comment]
Setting `connect-src` CSP directive and an appropriate Referrer-Policy header should be OK as a mitigation of potential future issues

**Comment by curphey** [comment]
Thats great. Alex did and a new library as well. Added you to a meeting at 1pm today so can we hopefully all check this and close it.
